File: GoLang/A02.ValidAnagram/valid_anagram_test.go
package valid_anagram

import (
	"testing"
)

func TestCode(t *testing.T) {
	var tests = []struct {
		s      string
		t      string
		output bool
	}{
		{
			s:      "rat",
			t:      "car",
			output: false,
		},
		{
			s:      "anagram",
			t:      "nagaram",
			output: true,
		},
	}

	for _, test := range tests {
		if got := ValidAnagram(test.s, test.t); got != test.output {
			t.Errorf(
				"For Strings %v and %v; Expected %v; Got %v",
				test.s, test.t, test.output, got,
			)
		}
	}
}

File: Python/01. Arrays and Hashing/02. Valid Anagram/valid_anagram_test.py
import pytest
from typing import List, NamedTuple

from valid_anagram import Solution


class InputTuple(NamedTuple):
    s: str
    t: str


@pytest.fixture
def get_fixtures():
    first_input = InputTuple(s="anagram", t="nagaram")
    first_output = True

    second_input = InputTuple(s="rat", t="car")
    second_output = False

    return [
        (first_input, first_output),
        (second_input, second_output),
    ]


def test_code(get_fixtures):
    for data in get_fixtures:
        assert Solution().execute(*data[0]) == data[1]
